bin/propolis-server/src/lib/vm/state_driver.rs

```diff
@@ -798,13 +830,99 @@ mod tests {
             .await
             .unwrap();
 
+        // The migration should appear to have finished. The state driver will
+        // queue a "stop" command to itself in this case, but because the driver
+        // is not directly processing the queue here, the test has to issue this
+        // call itself.
+        assert!(matches!(
+            driver.driver.get_migration_state(),
+            Some((_, ApiMigrationState::Finish))
+        ));
+        assert_eq!(
+            driver.driver.get_migration_state().unwrap().0,
+            migration_id
+        );
```

Comment:
You can fold these together with a guard on the `matches!`:

```suggestion
        assert!(matches!(
            driver.driver.get_migration_state(),
            Some((id, ApiMigrationState::Finish)) if id == migration_id
        ));
```

Reply:
Good call. Done in 3a2c071.